week3.js

```diff
@@ -0,0 +1,131 @@
+//String
+let myString = "Hello,this,is,a,difficult-,to-,read,sentence";
+console.log(myString);
+console.log(myString.length);
+myString = myString.replace(/,/g, " ");
+console.log(myString);
+
+//Arrays
+let favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+console.log(favoriteAnimals);
+
+let favoriteAnimalsOfMauro = ["blowfish", "capricorn", "giraffe", "turtle"];
+console.log(favoriteAnimalsOfMauro);
+
+let favoriteAnimalsOfJim = ["blowfish", "meerkat", "capricorn", "giraffe", "turtle"];
+console.log("The favorite animal of Jim is added to the array.");
+console.log(favoriteAnimalsOfJim);
+console.log("The array has a length of:", favoriteAnimalsOfJim.length);
+
+let favoriteAnimalsOfJason = ["blowfish", "meerkat", "capricorn", "turtle"];
+console.log(favoriteAnimalsOfJason);
+console.log(favoriteAnimalsOfJason.length);
+console.log(favoriteAnimalsOfJason[1], "The item you are looking for is at index:2");
+
+//JavaScript
+function sum(a, b, c) {
+    return a + b + c;
+}
+console.log(sum(5, 3, 2));
+//2.
+function colorCar(color){
+    return "A" + color + "car";
+ }
+ console.log(colorCar("red"));
+//3.
+let country = {Land: "The Netherlands", Population:"17 million", Capital:"Amsterdam"};
+console.log(country);
+console.log(country.Land);
+console.log(country.Population);
+//4.
+function vehicleType(color, code){
+    if (code === 1){
+    console.log("A" + color + "car");
+} else if (code === 2){
+console.log("A" + color + "motorbike");}
+}
+vehicleType("blue", 2);
+//5.
+3===3 ? console.log("Yes") : console.log("No");
+//6
+function vehicle(color, code, age){
+    if (code === 1 || age > 5){
+    console.log("A" + "" + color + "" + "used car");
+} else if (code === 2 || age < 5){
+    console.log("A" + "" + color + "" + "new car");
+}
+    }
+vehicle("blue", 1, 5);
+//7.
+const vehicleList = [
+     "motorbike",
+     "caravan",
+     "bike",
+     "bicycle",
+     "car"
+];
+console.log(vehicleList.length);
+console.log(vehicleList[2]);
+
+//9
+function newVehicle(color){
+    if (vehicleList[2] = 3){
+        console.log("A" + "" + color + "" + "new" + "" + "bike");
+    } else (vehicleList[2] > 3);{
+        console.log("A" + "" + color + "" + "old" + "" + "bike")
+    }
+}
+newVehicle("green", 3, 1);
+

```

Comment:
The  `newVehicle`  function needs to have three parameters, and you need to use those parameters to make your function reusable. Now your function uses 'bike' as the only vehicle type, what if we want to print  'an old black caravan ' or  ' a new red car'?

Reply:
Changed to: function newVehicle(color, code, age){
    if (vehicleList[2] <= 3){
        console.log("A" + " " + color + " " + "new" + " " + "bike");
    } else (vehicleList[2] > 3);{
        console.log("A" + " " + color + " " + "old" + " " + "bike")
    }
}
newVehicle("green", 3, 1);

Reply:
You have not used code and age parameters. This function still needs to be improved. You still use 'bike' as the only vehicle type